MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/ReflCommand.h

```diff
@@ -33,12 +35,27 @@ Code Documentation is available at: <http://doxygen.mantidproject.org>
 */
 class ReflCommand {
 public:
+  ReflCommand() : m_child(){};
   virtual ~ReflCommand(){};
 
   virtual void execute() = 0;
   virtual std::string name() = 0;
   virtual std::string icon() = 0;
+  virtual bool hasChild() final { return !m_child.empty(); };
+  virtual void setChild(std::vector<std::unique_ptr<ReflCommand>> child) final {
+    m_child = std::move(child);
+  }
+  virtual const std::vector<std::unique_ptr<ReflCommand>> &getChild() final {
+    return m_child;
+  }
+	virtual bool isSeparator() final { return name().empty() && icon().empty(); }
```

Comment:
I don't understand why this is declared both `virtual` and `final`. `final` has been around in other languages for many years and I can honestly say, I've never found an occasion where it was correct to use it. The `override` keyword is an excellent addition. What you are effectively doing with `final` is dictating to derived classes how they ought to implement things, that seems backwards to me.